modules/openapi-generator/src/main/resources/rust/Cargo.mustache

```diff
@@ -4,14 +4,21 @@ version = "{{{packageVersion}}}"
 authors = ["OpenAPI Generator team and contributors"]
 
 [dependencies]
-serde = "1.0"
-serde_derive = "1.0"
+serde = "^1.0"
+serde_derive = "^1.0"
+serde_json = "^1.0"
+hyper = "~0.11"
+url = "1.5"
+{{#hyper}}
 serde_yaml = "0.7"
-serde_json = "1.0"
 base64 = "~0.7.0"
-futures = "0.1.16"
-hyper = "0.11.6"
-url = "1.5"
+futures = "0.1.23"
+{{/hyper}}
+{{#reqwest}}
+reqwest = "~0.8"
+{{/reqwest}}
 
 [dev-dependencies]
+{{#hyper}}
 tokio-core = "*"
+{{/hyper}}
```

Comment:
My highest-level question - why make the decision at codegen time rather than at compile time? You could presumably achieve the same effect with [Rust features](https://doc.rust-lang.org/cargo/reference/manifest.html#the-features-section). I'm not saying you're wrong - merely curious as to why you chose as you did.

Reply:
I don't have a good reason for that. I am a beginner in Rust development and learning.

I don't know what purpose this dev lib deserves. Since I could generate my project without it in the Cargo file, I dropped it, simply to keep the file as tiny as possible.

If you think it is a mistake, I can put it back.

Reply:
I probably misunderstood your question (I thought it was about `tokio-core` lib only).

For the whole library choice, the main reason is that since I'm learning Rust, I don't know how to use it yet.

Another reason whould be that `hyper` and `reqwest` templates are quite different: API main methods currently don't have the same return type (`Result` vs `Box<Future<Result>>>`. So I presume that generating a client managing both libs with features would have much more complex code (but since I don't know features, I may be wrong).

Reply:
That seems reasonable. I could well imagine that trying to have both in the same crate could cause problems if they ever had incompatible dependencies.